src/inatinqperf/adaptors/weaviate_adaptor.py

```diff
@@ -1,257 +1,340 @@
-            "{\n  Get {\n    "
-            f"{self.class_name}(nearVector: {{ vector: {vector_json} }}, limit: {topk}) "
-            "{\n      originalId\n      _additional { id distance }\n    }\n  }\n}"
+        builder = (
+            self._client.query.get(self.class_name, [self._ORIGINAL_ID_FIELD])
+            .with_near_vector({"vector": vector.tolist()})
+            .with_limit(topk)
+            .with_additional(["id", "distance"])
         )
-        payload = {"query": query_str}
-        response = self._session.post(self._graphql_endpoint, json=payload, timeout=self.timeout)
 
-        if response.status_code != HTTPStatus.OK:
-            self._raise_error("search request failed", response)
+        if filters:
+            # Weaviate's where clause enables server-side filtering.
+            builder = builder.with_where(filters)
 
-        data = response.json()
-        results = self._extract_results(data)
-
-        search_results = []
-
-        for result in results[:topk]:
-            additional = result.get("_additional", {})
-            distance = float(additional.get("distance", "inf"))
-            id_str = additional.get("id")
-            original = result.get("originalId")
-            if original is not None:
-                data_id = int(original)
-
-            else:
-                try:
-                    data_id = self._validate_uuid(id_str) if id_str is not None else -1
-                except ValueError:
-                    data_id = -1
-
-            search_results.append(SearchResult(id=data_id, score=distance))
+        try:
+            data = builder.do()
+        except Exception as exc:  # pragma: no cover - defensive programming
+            status_code = getattr(exc, "status_code", None)
+            if status_code == HTTPStatus.UNPROCESSABLE_ENTITY:
+                logger.warning(
+                    f"[WeaviateAdaptor] search requested before schema exists (class={self.class_name})."
+                )
+                return []
+            self._handle_exception("search request failed", exc)
 
-        return search_results
+        results = self._extract_results(data)
+        return [SearchResult(id=result.id, score=result.score) for result in results[:topk]]
 
     def delete(self, ids: Sequence[int]) -> None:
-        """Delete objects for the provided identifiers."""
-        for identifier in ids:
-            obj_id = self._make_object_id(int(identifier))
-            response = self._session.delete(
-                f"{self._objects_endpoint}/{self.class_name}/{obj_id}", timeout=self.timeout
-            )
-            if response.status_code not in {200, 204, 404}:
-                self._raise_error("failed to delete object", response)
+        """Delete objects corresponding to the provided identifiers."""
+        object_ids = [self._make_object_id(int(identifier)) for identifier in ids]
+        for obj_id in object_ids:
+            self._delete_object(obj_id)
```

Comment:
Why are you doing a list comprehension above if we're going to loop here anyway?

You could simply do:
```python
for i in ids:
	self._delete_object(self._make_object_id(i))
```